doc/memory.rst

```diff
@@ -60,6 +60,13 @@ A simple example:
     Running f(2)
     2
 
+.. note::
+
+  :class:`~joblib.Memory` use pickle to load saved results from files, which
+  means that pickle security considerations apply, see `pickle documentation
+  <https://docs.python.org/3/library/pickle.html>`_ for more details. In
+  particular, only use a cache directory if you trust it.
```

Comment:
```suggestion
  :class:`~joblib.Memory` use pickle to load saved results from the file system.
  This means that pickle security considerations apply, see `pickle documentation
  <https://docs.python.org/3/library/pickle.html>`_ for more details.
  In particular, only use a cache directory if you trust it, and if there is a security
  concern about the filesystem, proper safeguards should be instanciated.
```

Reply:
Can you explain a bit what you mean by "proper safeguards should be instanciated" means :wink:.

Reply:
OK I removed the ending because I find it a bit too vague.

I have to admit I am not 100% happy with the wording but it's a start ...